integrations/mcp/src/haystack_integrations/tools/mcp/mcp_tool.py

```diff
@@ -234,19 +237,74 @@ async def call_tool(self, tool_name: str, tool_args: dict[str, Any]) -> str:
             message = "Not connected to an MCP server"
             raise MCPConnectionError(message=message, operation="call_tool")
 
-        try:
-            result = await self.session.call_tool(tool_name, tool_args)
-            if result.isError:
-                message = f"Tool '{tool_name}' returned an error: {result.content!s}"
-                raise MCPInvocationError(message=message, tool_name=tool_name)
-            return result.model_dump_json()
-        except MCPError:
-            # Re-raise specific MCP errors directly
-            raise
-        except Exception as e:
-            # Wrap other exceptions with context about which tool failed
-            message = f"Failed to invoke tool '{tool_name}' with args: {tool_args} , got error: {e!s}"
-            raise MCPInvocationError(message, tool_name, tool_args) from e
+        # Implement retry logic with exponential backoff for connection-related errors
+        for attempt in range(self.max_retries + 1):  # +1 for initial attempt
+            try:
+                result = await self.session.call_tool(tool_name, tool_args)
+                if result.isError:
+                    message = f"Tool '{tool_name}' returned an error: {result.content!s}"
+                    raise MCPInvocationError(message=message, tool_name=tool_name)
+                return result.model_dump_json()
+            except MCPError:
+                # Re-raise specific MCP errors directly (these are not connection issues)
+                raise
+            except (anyio.ClosedResourceError, ConnectionError, OSError) as e:
+                error_type = type(e).__name__
+                error_msg = str(e) if str(e) else "Connection closed unexpectedly"
+
+                # Don't retry on the last attempt
+                if attempt >= self.max_retries:
+                    message = (
+                        f"Tool '{tool_name}' failed after {self.max_retries} reconnection attempts. "
+                        f"Last error: {error_type}: {error_msg}. "
+                        f"Consider recreating the MCPTool instance or checking server availability."
+                    )
+                    raise MCPInvocationError(message, tool_name, tool_args) from e
+
+                # Only attempt reconnection for SSE/HTTP transports (if available)
+                if isinstance(self, SSEClient | StreamableHttpClient) and (
+                    sse_client is not None or streamablehttp_client is not None
+                ):
+                    logger.warning(f"Connection lost during tool call '{tool_name}': {error_type}: {error_msg}")
+
+                    try:
+                        # Exponential backoff before reconnection attempt
+                        if attempt > 0:
+                            delay = min(self.base_delay * (2 ** (attempt - 1)), 30)  # Cap at 30 seconds
```

Comment:
Would it make sense to make the 30 seconds hardcoded cap here as e.g. `max_delay` (between retries) in `__init__` (as we did for other vars)?